tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -86,25 +87,18 @@ public void CloseSpan(Span span)
         {
             bool ShouldTriggerPartialFlush() => Tracer.Settings.Exporter.PartialFlushEnabled && _spans.Count >= Tracer.Settings.Exporter.PartialFlushMinSpans;
 
-            if (span == RootSpan)
-            {
-                if (_samplingPriority == null)
-                {
-                    Log.Warning("Cannot set span metric for sampling priority before it has been set.");
-                }
-                else
-                {
-                    AddSamplingPriorityTags(span, _samplingPriority.Value);
-                }
-            }
-
             ArraySegment<Span> spansToWrite = default;
-
-            bool shouldPropagateMetadata = false;
+            var rootSpanInNextBatch = false;
 
             lock (this)
             {
                 _spans.Add(span);
+                if (!_rootSpanSent)
+                {
+                    _rootSpanInNextBatch |= (span == RootSpan);
```

Comment:
Once `_rootSpanInNextBatch` is set to true, it will never go back to being `false`, which seems like a bug. Maybe we can remove the field `_rootSpanInNextBatch` altogether and just assign this boolean result to the `rootSpanInNextBatch` local variable?